### ISSUE TITLE
feat(MatTab): Allow MatTab to be focusable when being disabled

### ISSUE THREAD

**Issue by clamli** [angular/components#26395]
### Feature Description

Currently a disabled MatTab is not focusable. In [Enabled tab vs Disabled tab demo](https://stackblitz.com/edit/angular-4shdxw-rygzeb?file=src/app/tab-group-animations-example.html), when we place the focus indicator on "Enabled tab 1" and use left/right arrow keys to navigate, we can see that "Disabled tab 1" is bypassed. 

This is fine for most cases, however, if there is a tooltip attached to the disabled tab which announces the reason why the tab is disabled, it would be hard for screen reader users to focus on it and read the message out. Because of that, [6.7 Focusability of disabled controls](https://w3c.github.io/aria-practices/#kbd_disabled_controls) encourages developers to keep "Tab elements in a set of Tabs" focusable when being disabled, for the purpose of accessibility.

### Use Case

We expect disabled tabs to be keyboard focusable, in details, 
- We want focus indicators to render appropriately on disabled tabs with [appropriate contrast ratio](https://www.w3.org/WAI/WCAG21/Understanding/contrast-minimum).
- We should consider making similar changes to other components mentioned in [6.7 Focusability of disabled controls](https://w3c.github.io/aria-practices/#kbd_disabled_controls) if applicable.
